added https support by loading certs from an external volume

kaestli opened the pull request:
exporting common nginx config for http and https to a snippet
updated documentation

kaestli committed:
added https support by loading certs from an external volume
exporting common nginx config for http and https to a snippet

damb commented on docs/docker-compose/README.rst:
```diff
@@ -54,6 +53,12 @@ preparation and/or configuration:
   ``eidaws-federator`` implements the actual federating service component.
 
+  If you plan to expose the federator via https, place the certificate, and the 
+  private key in /etc/ngingx/cert, under the following names:
```
Please fix the typo in the `/etc/nginx/cert`. Besides make sure to format the content properly. The markup format is  [ReStructuredText](https://en.wikipedia.org/wiki/ReStructuredText). Also, filenames should be formatted with `monospace`.

kaestli replied:
done

damb commented on docker-compose/docker-compose.yml:
```diff
@@ -14,13 +14,15 @@ services:
           nocopy: false
     ports:
-      - "8080:8080"
+      - "80:8080"
```
For testing purposes I'd rather set the ports to nonprivileged ports. On the machine actually deploying the stuff, I had a *secret* commit on top where the ports were adjusted to the privileged port values.

kaestli replied:
this would then apply to 443 as well, which is currently also mapped 1:1 ?

damb replied:
Exactly. Alternatively, provide a HTTPS specific *override* config (see e.g. https://docs.docker.com/compose/extends/). This way, the HTTPS related config is decoupled from the HTTP (*base*) config (regarding docker-compose this is IMO the cleaner approach).

Besides, it would be nice to decouple HTTP and HTTPS in the context of the `eidaws-federator` container image.

kaestli replied:
ack. ll do with an additional production compose file to overwrite standard ports

kaestli replied:
done

damb commented on docker-compose/federator/conf.d/http.conf:
```diff
@@ -15,135 +15,18 @@ http {
-      proxy_read_timeout 915s;
-      proxy_pass http://eida-federator-station-xml;
-    }
+    client_max_body_size 1M;
```
Is there a good reason, why the max size is configured when HTTPS enabled? The *federating instances* should have a max body size limit enabled, as well.

kaestli replied:
it is configured in both cases, http and https. Has been in place for http, now also for https

damb reviewed (changes requested):
@kaestli, thanks for this PR.

Some general comments from my side:
- When trying to start the system without the certificate available the `federator` service fails with something like
  ```
  $ docker logs eidaws-federator
  [...]
  nginx: [emerg] cannot load certificate "/etc/nginx/cert/eida-federator.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/cert/eida-federator.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
  [...]
  ```

  This is not ideal since for testing HTTPS should not be strictly required. Therefore, I'd favor a solution where HTTPS may be optionally deployed. What do you think?

- Also, the documentation added pretends to have the option to deploy the containers without HTTPS. In fact, at the time being, this is not the case. 
- Also, please update the `federator` image version to `eidaws-federator:1.2.0`

kaestli committed:
Merge branch 'EIDA:master' into https

kaestli committed:
modified https: one nginx config file for http or https; selection is done with a variable in the Docker file.
added an overwrite ...yml setting ports for production environments.

kaestli committed:
:q
Merge branch 'https' of https://github.com/kaestli/eidaws-federator-deployment into https

kaestli commented:
Will redo better :-)
